app/models/user.rb

```diff
@@ -18,7 +18,7 @@ class User < ActiveRecord::Base
   enum role: [:auditor, :accountant, :manager, :admin]
 
   ## Callbacks
-  after_create :make_admin!
+  after_commit :make_admin!
```

Comment:
But this method will called each time the user record get updated?

Neat solution is set default constraint in DB with value of admin. So whenever user gets created, he will be admin if role is not specified explicitly.